src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowEditBase.cs

```diff
@@ -181,34 +181,32 @@ protected WhereClause GetWhereClause(bool parameterize)
                 }
                 else
                 {
-                    if (cellData.RawObject is byte[] ||
-                        col.DbColumn.DataTypeName.Equals("TEXT", StringComparison.OrdinalIgnoreCase) ||
-                        col.DbColumn.DataTypeName.Equals("NTEXT", StringComparison.OrdinalIgnoreCase))
+                    if (parameterize)
                     {
-                        // Special cases for byte[] and TEXT/NTEXT types
-                        cellDataClause = "IS NOT NULL";
```

Comment:
Did you ever look into the history of this to try and figure out why this was done in the first place? It's such a weird thing to do that there might be a good reason to do it like this instead of something similar to what you're doing.

Reply:
https://github.com/microsoft/sqltoolsservice/commit/795eba3da6419a22c33fc04419648b3a942c7a80#diff-1f1765bd0d92616e0edf269c197b8422 

According to the first version of this class, it apparently existed from the beginning. I can only guess that it was done because using "=" did not work on those types exactly due to incompatibility and so this was a placeholder.

Reply:
This was how the code from SSMS was written. I can't remember all the details exactly, but I think it had something to do with the possibility of multiple gig values for binary/full-text columns.